sympy/matrices/matrices.py

```diff
@@ -1594,6 +1596,12 @@ def __pow__(self, num):
                     return self._matrix_pow_by_recursion(num)
             return self._matrix_pow_by_recursion(num)
         elif isinstance(num, (Expr, float)):
+            num = sympify(num)
+            if (num >= 0) == True and not num % 1 == 0:
+                z = self.jordan_form()[1]
```

Comment:
There should be no need for an alternative when the Jordan block method is used. In case recursion is used, and the exponent is negative, trying to compute the inverse will reveal singular matrices.